File: C#/GameWorld/GameWorld.cs
public class GameWorld
{
    public AABB aabb;
    public Boid[] boids = new Boid[300];

    public GameWorld()
    {
        var subgen = new Subgen(0);
        aabb = new AABB(MyVector2.Zero, new MyVector2(50,50));
        for (int i = 0; i < boids.Length; i++)
            boids[i] = new Boid(aabb,subgen);
    }
    public void Update(long deltaTime)
    {
        var deltaTimeInSeconds = ((float)deltaTime)/1000f;
        Boid.Update(boids, ref aabb, deltaTimeInSeconds);
    }
}

File: C/src/GameWorld.c
#include "Boid.h"

#define GAMEWORLD_BOIDSCOUNT 300

typedef struct GameWorld
{
    AABB aabb;
    Boid boids[GAMEWORLD_BOIDSCOUNT];
} GameWorld;

GameWorld GameWorld_Create()
{
    GameWorld gameWorld;

    Subgen subgen = Subgen_Init(0);
    gameWorld.aabb.p0 = (MyVector2){ 0, 0 };
    gameWorld.aabb.p1 = (MyVector2){ 50, 50 };
	for (int i = 0; i < GAMEWORLD_BOIDSCOUNT; i++)
		gameWorld.boids[i] = Boid_Create(&gameWorld.aabb,&subgen);

    return gameWorld;
}
void GameWorld_Update(GameWorld* gameWorld, long deltaTimeInMilliseconds)
{
    float deltaTimeInSeconds = ((float)deltaTimeInMilliseconds)/1000;
    Boid_Update(gameWorld->boids,GAMEWORLD_BOIDSCOUNT,&gameWorld->aabb,deltaTimeInSeconds);
}